### PR TITLE
Have `request.url` take `relative_url_root` into account

### PR DESCRIPTION
**Pull request by maclover7**
ref #24393

### CODE REVIEW

File: actionpack/lib/action_dispatch/http/url.rb [hunk 1]
```diff
@@ -372,6 +372,12 @@ def subdomains(tld_length = @@tld_length)
       def subdomain(tld_length = @@tld_length)
         ActionDispatch::Http::URL.extract_subdomain(host, tld_length)
       end
+
+      private
+
+      def relative_url_root
+        get_header('action_controller.instance'.freeze).relative_url_root || ''
```
Comment (vipulnsward) [inline actionpack/lib/action_dispatch/http/url.rb]: ''.freeze

**Comment by pixeltrix** [comment]
I have a terrible feeling that doing this will break loads of applications

**Comment by guilleiguaran** [comment]
I believed that we deprecated `RAILS_RELATIVE_URL_ROOT` in favor of Rack's `SCRIPT_NAME` years ago but now I see that we never did it.

**Comment by pixeltrix** [comment]
@guilleiguaran unfortunately `SCRIPT_NAME` doesn't exist outside of a request so wouldn't work for mailers, background jobs, etc. This is why we un-deprecated `RAILS_RELATIVE_URL_ROOT`.

**Comment by kikonen** [comment]
too bad that this was closed

**Pull request closed without merging**